cmd/kyma/alpha/provision/k3s/cmd.go

```diff
@@ -114,12 +115,17 @@ func (c *command) deleteExistingK3sCluster() error {
 }
 
 //Check if a port is allocated
-func (c *command) portAllocated(port int) bool {
-	con, err := net.Listen("tcp", fmt.Sprintf(":%d", port))
-	if con != nil {
-		con.Close()
+func (c *command) portAllocated(ports ...int) error {
```

Comment:
I would rename this function to something different if it no longer returns a boolean.
For example `allocatePorts`.